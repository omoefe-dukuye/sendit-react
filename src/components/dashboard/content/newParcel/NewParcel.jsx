import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

const NewParcel = ({ parcel: {
  id,
  current_location: location,
  destination,
  distance,
  price
} }) => (
  <div className='create-order__new-order'>
    <h3>New Order Created</h3>
    <ul>
      <li>Parcel ID: {id}</li>
      <li>Location: {location}</li>
      <li>Destination: {destination}</li>
      <li>Distance: {distance}</li>
      <li>Price: {price}</li>
    </ul>
  </div>
);

NewParcel.propTypes = {
  parcel: PropTypes.object
};

export default NewParcel;
