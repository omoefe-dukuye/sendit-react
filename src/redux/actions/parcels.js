import { toast } from "react-toastify";
import axios from "../../utils/axiosConfig";
import { ADD_ORDER } from "./types";

const api = process.env.API_ROOT_URL;

export const createOrder = (payload) => async (dispatch) => {
  try {
    const { data: { parcel } } = await axios.post(`${api}/parcels`, payload);

    dispatch({
      type: ADD_ORDER,
      payload: parcel
    });

    return parcel;
  } catch ({ response: { data: { error } } }) {
   toast.error(error);
  }
}

export default createOrder;
