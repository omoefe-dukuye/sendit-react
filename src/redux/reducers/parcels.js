import { ADD_ORDER } from '../actions/types';

const parcelReducerDefaultState = [];

export default (state = parcelReducerDefaultState, action) => {
  switch(action.type) {
    case ADD_ORDER:
      return [...state, action.payload];

    default:
      return state;
  }
};
