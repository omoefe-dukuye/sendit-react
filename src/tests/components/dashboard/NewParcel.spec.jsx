import React from 'react';
import { shallow } from 'enzyme';
import '@babel/polyfill';
import NewParcel from '../../../components/adminDashboard/content/newParcel';
import parcels from '../../mocks/mockParcels';

test('Should render static component', () => {
  const wrapper = shallow(<NewParcel parcel={parcels[0]} />);
  expect(wrapper).toMatchSnapshot();
});
