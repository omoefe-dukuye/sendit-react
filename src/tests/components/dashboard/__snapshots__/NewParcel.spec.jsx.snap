// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render static component 1`] = `
<div
  className="create-order__new-order"
>
  <h3>
    New Order Created
  </h3>
  <ul>
    <li>
      Parcel ID: 
      34
    </li>
    <li>
      Location: 
      1 Aminu St, Mende, Lagos, Nigeria
    </li>
    <li>
      Destination: 
      Mushin, Lagos, Nigeria
    </li>
    <li>
      Distance: 
      5
    </li>
    <li>
      Price: 
      N 625
    </li>
  </ul>
</div>
`;
